prov/usnic/src/usdf_cq.c

```diff
@@ -119,7 +119,55 @@ usdf_cq_readerr(struct fid_cq *fcq, struct fi_cq_err_entry *entry,
 
 	cq->cq_comp.uc_status = 0;
 
-	return sizeof(*entry);
+	return 1;
+}
+
+static ssize_t
+usdf_cq_readerr_soft(struct fid_cq *fcq, struct fi_cq_err_entry *entry,
+		uint64_t flags)
+{
+	struct usdf_cq *cq;
+	size_t soft_cq_entry_len;
+	struct usdf_cq_soft_entry *soft_comp_entry;
+	void *tail;
+
+	USDF_TRACE_SYS(CQ, "\n");
+
+	cq = container_of(fcq, struct usdf_cq, cq_fid);
+
+	// The return values are analogous to sockets cq_readerr
+	tail = cq->c.soft.cq_tail;
+	soft_cq_entry_len = sizeof(struct usdf_cq_soft_entry);
+	soft_comp_entry = (struct usdf_cq_soft_entry *)tail;
+
+	entry->op_context = soft_comp_entry->op_context;
+	entry->flags = 0;
+	entry->err = FI_EIO;
+	switch (soft_comp_entry->status) {
+	case USD_COMPSTAT_SUCCESS:
```

Comment:
I don't think we should be using `USD_COMPSTAT_` statuses here, since the "thing" that was completing never directly came from usnic_direct.